### PR TITLE
perf(831): Remove row_pk computation from query

### LINKED ISSUE #280: Remove PrimaryKey
The current `PrimaryKey` type is just a wrapper for `DataKey` which uniquely identifies each row in the database. It is not a column or row attribute and therefore does correspond to a traditional primary key in a relational database. We should remove to avoid confusion.

### LINKED ISSUE #831: Subscriptions should not compute primary keys
The new datastore no longer stores a `DataKey` with each row. This means that the `id` field of a `RelValue` is always `None`, which in turn means that we will compute a `PrimaryKey` for each row returned in a subscription. This dominates the cost of subscription evaluation.

The following graphs were generated from a table scan of 1M fixed length rows. Here `PrimaryKey` represents ~50% of the total execution time. And the relative cost only grows as more rows are added to the result set.

Image

Image

Clients use the `PrimaryKey` to delete rows from their local cache. But in most cases hashing the row itself will be no more expensive than hashing the `PrimaryKey`.

### LINKED ISSUE #874: Remove `RelationalDB::pk_for_row`
See e.g., https://github.com/clockworklabs/SpacetimeDB/pull/869/files#r1497828101 and friends.

### PR DESCRIPTION
**Pull request by joshua-spacetime**
Closes #831.
Closes #874.
Closes #280.

```
full-scan               time:   [105.90 ms 106.18 ms 106.51 ms]
                        change: [-53.505% -53.317% -53.150%] (p = 0.00 < 0.05)
                        Performance has improved.

full-join               time:   [312.20 µs 313.41 µs 314.63 µs]
                        change: [-33.453% -33.176% -32.873%] (p = 0.00 < 0.05)
                        Performance has improved.

incr-select             time:   [2.0535 µs 2.0584 µs 2.0649 µs]
                        change: [-12.335% -12.142% -11.936%] (p = 0.00 < 0.05)
                        Performance has improved.

incr-join               time:   [10.726 µs 10.747 µs 10.769 µs]
                        change: [+0.3173% +0.5243% +0.7765%] (p = 0.00 < 0.05)
                        Change within noise threshold.
```

Removes some `#[tracing::instrument(skip_all)]` as well.

# Expected complexity level and risk

1

### CODE REVIEW

File: crates/core/src/subscription/query.rs [hunk 1]
```diff
@@ -404,46 +392,6 @@ mod tests {
         Ok(())
     }
 
-    #[test]
```
Comment (cloutiertyler) [inline crates/core/src/subscription/query.rs:407]: Maybe we want to check that the row values are equal?

File: crates/core/src/subscription/module_subscription_actor.rs [hunk 2]
```diff
@@ -89,6 +89,7 @@ impl ModuleSubscriptions {
         // thread it's possible for messages to get sent to the client out of order. If you do
         // spawn in another thread messages will need to be buffered until the state is sent out
         // on the wire
+        // let _span = tracing::info_span!("send").entered();
```
Comment (cloutiertyler) [inline crates/core/src/subscription/module_subscription_actor.rs:92]: Maybe pull this out?

**Review by cloutiertyler — COMMENTED** [review]
This seems correct to me. Just two comments. Also wow, that scan speedup.

File: crates/core/src/subscription/module_subscription_actor.rs (hunk 2 above)
Comment (joshua-spacetime) [reply crates/core/src/subscription/module_subscription_actor.rs:92]: Ah yes, some left over debugging.

File: crates/core/src/subscription/query.rs (hunk 1 above)
Comment (joshua-spacetime) [reply crates/core/src/subscription/query.rs:407]: Yes, but technically we already have tests for that.

**Pull request merged**